### ISSUE TITLE
Confusing documentation for AggregateException.GetBaseException(...)

### ISSUE THREAD

**Issue by kmcclellan** [dotnet/runtime#82853]
The [comment for `AggregateException.GetBaseException()`](https://github.com/dotnet/runtime/blob/0a15c3b97dabaf94e5d4399aef2fd48c2b3f446a/src/libraries/System.Private.CoreLib/src/System/AggregateException.cs#L215) says it returns an `AggregateException`, which is not always (or even usually) the case.

**Comment by danmoseley** [comment]
@skyoxZ your implementations above are functionally equivalent or am I missing something? they just use the stack.

@kmcclellan I agree the docs and comment doesn't match the implementation. It seems to me the implementation is right. Do you want to offer a PR to fix the comment (and add a test -- https://github.com/dotnet/runtime/blob/f70d8a9663d3d674c453e26d405dc13baa79fd16/src/libraries/System.Threading.Tasks/tests/AggregateExceptionTests.cs#L39-L55) ? The docs are in another repo, there's an edit button on the [doc page](https://learn.microsoft.com/en-us/dotnet/api/system.aggregateexception.getbaseexception?view=net-8.0)

**Comment by skyoxZ** [comment]
> Find the deepest exception, but stop in the case where there's more than one to choose from

@danmoseley Current behavior is not:

1. `Exception.GetBaseException()` always digs to `InnerException` even if reaching an `AggregateException` with more than 1 innerExceptions.

2. `AggregateException.GetBaseException()` will not dig more when its only inner exception is not an `AggregateException` if even if there's deeper exception.

**Comment by danmoseley** [comment]
Ah, yes. I suspect this would be unacceptably breaking. @stephentoub ?

**Comment by stephentoub** [comment]
> Ah, yes. I suspect this would be unacceptably breaking. @stephentoub ?

Yeah, I don't think there's a compelling enough ROI for such a change.

**Comment by skyoxZ** [comment]
yes, their `ToString()`'s include the whole inner-exception tree.

**Comment by tannergooding** [comment]
Given @stephentoub's comment, it sounds like this needs to be a simple docs change to call out the actual behavior.
